### ISSUE TITLE
Rescale to zero torch.nn.CrossEntropyLoss with label_smoothing > 0

### ISSUE THREAD

**Issue by AdrienCourtois** [pytorch/pytorch#130342]
### 🚀 The feature, motivation and pitch

Currently, the cross entropy loss using `label_smoothing > 0` does not reach its minimum when the loss is zero. This can become problematic when you try to compare two models with different `label_smoothing` using the loss values. It is also quite tricky when the model converges well because we can't measure the convergence using the logarithm of the loss.

Instead, the minimum is given by
$$-(1 - \alpha) \log(1-\alpha) - \alpha (\log(\alpha) - \log(K-1)),$$
where $\alpha$ is `label_smoothing` and $K$ is the total number of classes.

For $\alpha=0.1$ and $K=1000$, the minimum is equal to approximatively $4.4 \times 10^{-1}$.

### Alternatives

Practitioners can fix this problem themselves by defining a new CrossEntropyLoss and rescaling the loss themselves.

**Comment by mikaylagawarecki** [comment]
Could you clarify what you mean by 

> the cross entropy loss using label_smoothing > 0 does not reach its minimum when the loss is zero

Does this mean that when the loss is 0 the label does not match the target perfectly. This seems like the expected behavior since label smoothing adds a uniform distribution to the target

**Comment by AdrienCourtois** [comment]
It means that the loss never reaches 0. In the best case *i.e.* when the predicted probability is equal to the correct label + the uniform distribution, the loss is equal to the value noted in the first message. 
A simple rescaling would ensure the loss can reach 0 when the network converges to the minimal value.

**Comment by jbschlosser** [comment]
> Practitioners can fix this problem themselves by defining a new CrossEntropyLoss and rescaling the loss themselves.

If you specifically want the property that the loss can go to zero, I do think it's best to define a custom loss function. As-is, we implemented label smoothing according to the description in the [paper](https://arxiv.org/abs/1512.00567) in which it was originally introduced. For the `label_smoothing` arg in `torch.nn.CrossEntropyLoss`, I think we only want to introduce a fix if we diverge somehow from the implementation described there.

**Comment by AdrienCourtois** [comment]
While I agree with your reasoning, I might argue that (correct me if I’m wrong) this is the only loss in the library that doesn’t reach its minimum at zero.
